### PR TITLE
added phone number generator for e164

### PR DESCRIPTION
**Pull request by sandrom**
As proposed in #40 added a phone number generator. There are preferred writings of phone numbers, but I have opted to add some random, for better testing.

here some examples, that were generated:
+62 8 0354110493 0
+974 39 721044551
+233 007 4 41 36014
+967 5 228269819 2
+992 02 83 89590 8 8
+299 83 5 8044 4 7 7 4
+39 12 1845350 41 7
+250 51 1193 783 6 0
+1 758 8 212562 7 25
+239 2 840073 7266
+230 207 33333 710
+594 620 5 26180 29
+689 4 08876 1 802 8
+500 7 736772 39 37
+46 34 06622194 7 7
+870 515 4 202 119 4
+597 0 299646601 2
+973 7 1657347420
+43 654 799 611 01 5
+376 23 52 75 64833
+1 670 4 5 4693995 8
+1 670 922 664 1 7 7 6
+381 917 76 28 65 27
+960 3 036 1 2 8290 6
+672 819 45 794671
+36 52 81 083 3 2792
+222 116 9130972 2
+216 8 783530 5 68 2
+221 18 4 221983 5 5
+61 16 97473218 1 2
+968 78 0577882 53
+264 2 90057716 7 6
+375 8 505 78 50116
+227 5 20877584 8 5
+357 18 1017 2 096 4
+690 35 06 890018 4
+505 421 6001128 3
+43 86 2 888 0 431 1 3
+965 89 912577094
+351 13 848 4 8011 4
+353 46 5106954 4 6
+685 6 391221 88 0 8
+251 560 3300 7 784
+599 5 805932034 7
+385 220 0 19 1384 2
+355 32 067 0 52235
+291 0 073 67886 75
+590 639 22 06 574 3
+690 727 016507 6 5
+238 91 5866840 6 2

### CODE REVIEW

**Comment by Pallinder** [comment]
Nice! Thanks for the pr, had  a few comments that I posted.

**Comment by Pallinder** [comment]
Also, please add documentation so that people know the feature is there.

File: jsondata.go [hunk 1]
```diff
@@ -2612,6 +2612,9 @@ var data = []byte(`{
         "W-SU",
         "Zulu"
     ],
+    "callingCodes" : [
```
[the remark this answers is not in the document]
Comment (sandrom) [reply jsondata.go]: https://en.wikipedia.org/wiki/List_of_country_calling_codes

actually i found it often in the internet just named "calling code" without country, so I opted for that name as it seems to be quite correct. If you wish we could change it to countryCallingCodes, but as said, also dont wanna overdo it with the java-names :D

File: random_data_test.go [hunk 2]
```diff
@@ -422,3 +422,11 @@ func findInSlice(source []string, toFind string) bool {
 	}
 	return false
 }
+
+func TestPhoneNumbers(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sandrom) [reply random_data_test.go]: wasnt sure how far I should go here as it would double already small code amount but you're probably right here

**Comment by sandrom** [comment]
I was contemplating putting a country-callingcode mapping in there but it made things much more complex and I did not want to blow up this mapping too much in this iteration. There is actually some specifics to some countries when it comes to real phone numbers on how they are usually written, but that should not stop those generated here from still being valid anyway. For the time being it was really to generate international ones, so testing with numbers, that may not fall into your normal local categories also happens. I think that is probably more important at the end of the day, that it stays flexible enough. It could be made more complex though if necessary

File: jsondata.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pallinder) [reply jsondata.go]: Hehe, I don't think that's to java. I think countryCallingCodes is clearer in what it actually contains.

File: random_data_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Pallinder) [reply random_data_test.go]: No worries about code amount on the tests.

**Comment by Pallinder** [comment]
Let's just do the rename and we are good to go!

**Comment by sandrom** [comment]
yes sorry, I was in holiday, just came back from that :)

**Comment by Pallinder** [comment]
LGTM. Thanks!

**Pull request merged**